src/python/pants/build_graph/address.py

```diff
@@ -304,7 +338,10 @@ def to_address(self) -> Address:
         """Convert this BuildFileAddress to an Address."""
         # This is weird, since BuildFileAddress is a subtype of Address, but the engine's exact
         # type matching requires a new instance.
-        # TODO: Possibly BuildFileAddress should wrap an Address instead of subclassing it.
+        # TODO: Possibly create a new class like `AddressWithBuild` that wraps an `Address`. This
+        #  is weird to subclass `Address` but break Liskov substitution in many places, like the
```

Comment:
AFAIK, constructors are generally excluded from a Liskov sense. Still agree with the general gist though.